### ISSUE TITLE
When to Use Context

### ISSUE THREAD

**Issue by umarzirgham** [reactjs/reactjs.org#1551]
class ThemedButton extends React.Component {
  // Assign a contextType to read the current theme context.
  // React will find the closest theme Provider above and use its value.
  // In this example, the current theme is "dark".
  static contextType = ThemeContext;
  render() {
    return <Button theme={this.context} />;<strong>/*if I am not wrong here should be "this.contextType"*/</strong>
  }
}

**Comment by ashu8912** [comment]
No,its absolutely correct it should be this.context and not this.contextType.
contextType is a static property on the class ThemedButton meaning it can't be accessed with "this"
there are two ways of assigning "this.context"  
1.ThemedButton.contextType=ThemeContext
2.assigning it to the static property contextType 
In both of the above ways you will get your context on "this.context"


Hope you got it,,,,if not tell me i can further explain :)

**Comment by umarzirgham** [comment]
So do I consider this.context is reserved variable when we are using Context?

**Comment by ashu8912** [comment]
"this.context "is there not only when using context but it is a property that your class gets because of extending React.Component 
you can check this by simply 
saying 
console.log(this);
in your smart components(those which have inherited the React.Component class) render method.

you will find a context property to exist there by default which is initialised to an empty object;


when in our code we say 
static contextType = ThemeContext;
we are actually changing the already present this.context object

If its still not clear i can help :)

**Comment by umarzirgham** [comment]
Okay I got it now. Thanks for the clarification. Really appreciate. 👍
